### PR TITLE
Added p2p multikey support

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added p2p multikey support

reference directory: https://github.com/ElrondNetwork/elrond-go/tree/feat/multisigner/p2p
or compare (check p2p directory changes) https://github.com/ElrondNetwork/elrond-go/compare/rc/v1.4.0...feat/multisigner

### CODE REVIEW

File: libp2p/crypto/identityGenerator.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		return nil, "", err
+	}
+
+	return skBuff, core.PeerID(pid), nil
+}
+
+// CreateP2PPrivateKey will create a new P2P private key based on the provided seed. If the seed is the empty string
+// it will use the crypto's random generator to provide a random one. Otherwise, it will create a deterministic private
+// key. This is useful when we want a private key that never changes, such as in the network seeders
+func (generator *identityGenerator) CreateP2PPrivateKey(privateKeyBytes []byte) (libp2pCrypto.PrivKey, error) {
```
Comment (ssd04) [inline libp2p/crypto/identityGenerator.go:48]: update this comment?

File: libp2p/mockMessenger.go [hunk 2]
```diff
@@ -25,11 +26,11 @@ func NewMockMessenger(
 	}
 
 	ctx, cancelFunc := context.WithCancel(context.Background())
-	privKey, _ := createP2PPrivateKey(args.P2pPrivateKeyBytes)
+	keyGen := crypto.NewIdentityGenerator()
+	p2pPrivateKey, err := keyGen.CreateP2PPrivateKey(args.P2pPrivateKeyBytes)
+	signer, _ := crypto.NewP2PSigner(p2pPrivateKey)
```
Comment (ssd04) [inline libp2p/mockMessenger.go:31]: check errors here?

File: libp2p/netMessenger.go [hunk 3]
```diff
@@ -208,10 +207,14 @@ func constructNode(
 		return nil, err
 	}
 
+	p2pSignerInstance, err := crypto.NewP2PSigner(p2pPrivateKey)
+	if err != nil {
+		cancelFunc()
```
Comment (sstanculeanu) [inline libp2p/netMessenger.go:212]: why not move L203 on L215 and delete `cancelFunc()` here and on L206?

File: libp2p/crypto/identityGenerator.go (hunk 1 above)
Comment (iulianpascalau) [reply libp2p/crypto/identityGenerator.go:48]: done

File: libp2p/mockMessenger.go (hunk 2 above)
Comment (iulianpascalau) [reply libp2p/mockMessenger.go:31]: done

File: libp2p/netMessenger.go (hunk 3 above)
Comment (iulianpascalau) [reply libp2p/netMessenger.go:212]: great suggestion. Done

**Review by sstanculeanu — APPROVED** [review]

**Pull request merged**